src/Web3/Util.elm

```diff
@@ -0,0 +1,279 @@
+        , expect = expectJson hexDecoder
+        , callType = Sync
+        }
+
+
+toAscii : Hex -> Task Error String
+toAscii (Hex val) =
+    toTask
+        { func = "toAscii"
+        , args = Encode.list [ Encode.string val ]
+        , expect = expectJson toAsciiDecoder
+        , callType = Sync
+        }
+
+
+fromAscii : String -> Task Error Hex
+fromAscii val =
+    fromAsciiPadded 0 val
+
+
+fromAsciiPadded : Int -> String -> Task Error Hex
+fromAsciiPadded padding val =
+    toTask
+        { func = "fromAscii"
+        , args = Encode.list [ Encode.string val, Encode.int padding ]
+        , expect = expectJson hexDecoder
+        , callType = Sync
+        }
+
+
+toDecimal : Hex -> Task Error Int
+toDecimal (Hex hex) =
+    toTask
+        { func = "toDecimal"
+        , args = Encode.list [ Encode.string hex ]
+        , expect = expectInt
+        , callType = Sync
+        }
+
+
+fromDecimal : Int -> Task Error Hex
+fromDecimal decimal =
+    toTask
+        { func = "fromDecimal"
+        , args = Encode.list [ Encode.int decimal ]
+        , expect = expectJson hexDecoder
+        , callType = Sync
+        }
+
+
+isAddress : Address -> Task Error Bool
+isAddress (Address address) =
+    toTask
+        { func = "isAddress"
+        , args = Encode.list [ Encode.string address ]
+        , expect = expectBool
+        , callType = Sync
+        }
+
+
+isChecksumAddress : Address -> Task Error Bool
+isChecksumAddress (Address address) =
+    toTask
+        { func = "isChecksumAddress"
+        , args = Encode.list [ Encode.string address ]
+        , expect = expectBool
+        , callType = Sync
+        }
+
+
+toChecksumAddress : Address -> Task Error Address
+toChecksumAddress (Address address) =
+    toTask
+        { func = "toChecksumAddress"
+        , args = Encode.list [ Encode.string address ]
+        , expect = expectJson addressDecoder
+        , callType = Sync
+        }
+
+
+toWei : EthUnit -> String -> Result String BigInt
+toWei unit amount =
+    -- check to make sure input string is formatted correctly, should never error in here.
+    if Regex.contains (Regex.regex "^\\d*\\.?\\d+$") amount then
+        let
+            decimalPoints =
+                decimalShift unit
+
+            formatMantissa =
+                String.slice 0 decimalPoints >> String.padRight decimalPoints '0'
+
+            finalResult =
+                case (String.split "." amount) of
+                    [ a, b ] ->
+                        a ++ (formatMantissa b)
+
+                    [ a ] ->
+                        a ++ (formatMantissa "")
+
+                    _ ->
+                        "ImpossibleError"
+        in
+            case (BigInt.fromString finalResult) of
+                Just result ->
+                    Ok result
+
+                Nothing ->
+                    Err "There was an error calculating the result. However, the fault is not yours; please report this bug on github."
```

Comment:
Well this error technically should never happen, and if it does (which is impossible since I first check that the input string is correctly formated with regex) then they must definitely file a bug report.

Reply:
Well this error technically should never happen, and if it does (which is impossible since I first check that the input string is correctly formated with regex) then they must definitely file a bug report. But sure descriptive=good...